app/src/bcsc-theme/api/hooks/useRegistrationApi.tsx

```diff
@@ -113,34 +114,84 @@ const useRegistrationApi = (apiClient: BCSCApiClient | null, isClientReady: bool
   }, [isClientReady, apiClient, logger, dispatch])
 
   const updateRegistration = useCallback(
-    async (clientId: string) => {
-      if (!isClientReady || !apiClient) {
-        throw new Error('BCSC client not ready for registration update')
-      }
-
-      let fcmDeviceToken = 'fallback_fcm_token'
-      let apnsToken: string | null = ''
-
-      try {
-        const tokens = await getNotificationTokens()
-        fcmDeviceToken = tokens.fcmDeviceToken || fcmDeviceToken
-        apnsToken = tokens.apnsToken || apnsToken
-      } catch (error) {
-        // Log warning but continue with fallback tokens
-        logger.warn(
-          `Failed to retrieve tokens for registration update: ${error instanceof Error ? error.message : String(error)}`
-        )
-      }
-
-      const body = await getDynamicClientRegistrationBody(fcmDeviceToken, apnsToken || '')
-      const { data } = await apiClient.put<RegistrationResponseData>(
-        `${apiClient.endpoints.registration}/${clientId}`,
-        body,
-        { headers: { 'Content-Type': 'application/json' } }
-      )
-      return data
+    async (registrationAccessToken: string | undefined, selectedNickname: string | undefined) => {
+      return withAccount(async (account) => {
+        if (!isClientReady || !apiClient) {
+          throw new Error('BCSC client not ready for registration update')
+        }
+
+        if (!registrationAccessToken) {
+          throw new Error('No registration access token found for registration update')
+        }
+
+        if (!selectedNickname) {
+          throw new Error('No client name found for registration update')
+        }
+
+        let fcmDeviceToken = 'fallback_fcm_token'
+        let apnsToken = ''
+
+        try {
+          const tokens = await getNotificationTokens(logger)
+          fcmDeviceToken = tokens.fcmDeviceToken ?? fcmDeviceToken
+          apnsToken = tokens.apnsToken ?? apnsToken
+        } catch (error) {
+          // Log warning but continue with fallback tokens
+          logger.warn(
+            `Failed to retrieve tokens for registration update: ${
+              error instanceof Error ? error.message : String(error)
+            }`
+          )
+        }
+
+        const body = await getDynamicClientRegistrationBody(fcmDeviceToken, apnsToken)
+
+        let updatedRegistrationData: RegistrationResponseData | null = null
+
+        try {
+          const updatePayload = typeof body === 'string' ? JSON.parse(body) : body
```

Comment:
Curious why this is needed, were you sometimes getting an object back from `getDynamicClientRegistrationBody` and sometimes getting a string?

Reply:
`getDynamicClientRegistrationBody` can return either string or null. This can be cleaned up as its not specific to needing a type check but can just check if it exists.